hrm/utils/LogUtility.py

```diff
@@ -0,0 +1,114 @@
+
+    def info(self, msg):
+        fname, line_num = self.__get_call_info()
+        msg = "[{}:{}] - {}".format(fname, line_num, msg)
+        self.__logger.info(msg)
+
+    def warn(self, msg):
+        fname, line_num = self.__get_call_info()
+        msg = "[{}:{}] - {}".format(fname, line_num, msg)
+        self.__logger.warn(msg)
+        self.__logger.warn(traceback.format_exc())
+
+    def error(self, msg, trace=True):
+        fname, line_num = self.__get_call_info()
+        msg = "[{}:{}] - {}".format(fname, line_num, msg)
+        self.__logger.error(f"Error - {msg}")
+        if trace and (traceback.format_exc() != 'NoneType: None\n'):
+            self.__logger.error(traceback.format_exc())
+
+    def debug(self, msg, trace=True):
+        fname, line_num = self.__get_call_info()
+        msg = "[{}:{}] - {}".format(fname, line_num, msg)
+        self.__logger.debug(f"Debug - {msg}")
+        if trace and (traceback.format_exc() != 'NoneType: None\n'):
+            self.__logger.debug(traceback.format_exc())
+
+    @staticmethod
+    def get_logger(test_suite_name=None, log_level='INFO'):
+        if not LogUtility.__test_logger:
+            return LogUtility(test_suite_name, level=log_level)
+        else:
+            return LogUtility.__test_logger
+
+    def _get_logger(self, test_suite_name, level='INFO'):
+        """default python logger"""
+        accepted_levels = {'INFO': logging.INFO, 'DEBUG': logging.DEBUG}
+        if level.upper() not in accepted_levels:
+            raise Exception('INVALID LOG LEVEL SPECIFIED')
+        logging_format = '[%(asctime)s] %(levelname)s - %(message)s'  # {%(filename)s:%(lineno)d} %(asctime)s %(levelname)s: %(message)s'
+        date_format = '%Y-%m-%d %H:%M:%S'
+        test_log_directory = os.path.join(FolderUtility.get_log_directory(), test_suite_name)
+        FolderUtility.make_directory(test_log_directory)
+        tmp_log_file_folder = os.path.join(test_log_directory, "run_")
+        self.test_log_folder = FolderUtility.make_director_with_timestamp(tmp_log_file_folder)
+        log_file_path = os.path.join(self.test_log_folder, "test_.log")
+        test_log_path = FolderUtility.get_unique_file_path(log_file_path)
+        base_name, extension = os.path.splitext(test_log_path)
+        #debug_log_path = FolderUtility.get_unique_file_path(base_name + "_debug_" + extension)
+        logging.basicConfig(level=logging.NOTSET, format=logging_format, datefmt=date_format)
+        formatter = logging.Formatter(fmt=logging_format, datefmt=date_format)
+        handler = logging.FileHandler(test_log_path)
+        handler.setLevel(accepted_levels[level])
+        handler.setFormatter(formatter)
+        # ch = logging.StreamHandler()
+        # ch.setLevel(logging.INFO)
+        # ch.setFormatter(formatter)
+        #debug_handler = logging.FileHandler(debug_log_path)
+        #debug_handler.setLevel(logging.DEBUG)
+        #debug_handler.setFormatter(formatter)
```

Comment:
For code cleanliness if this is not required it should be removed

Reply:
Removed unused code.